extensions/wikia/PageHeader/PageTitle.class.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace PageHeader;
+
+use \RequestContext;
+
+class PageTitle
+{
+
+    /* @var string */
+    public $title;
+
+    /* @var string */
+    public $prefix;
+
+    /* @var \WikiaGlobalRegistry */
+    private $wg;
+
+    /* @var int */
+    private $namespace;
+
+    const PREFIX_LESS_NAMESPACES = [NS_MEDIAWIKI, NS_TEMPLATE, NS_CATEGORY, NS_FILE];
+
+    /**
+     * @param \WikiaApp $app
+     */
+    public function __construct($app)
+    {
+        $this->wg = $app->wg;
+        $this->MWTitle = RequestContext::getMain()->getTitle();
+
+        $this->namespace = $this->MWTitle->getNamespace();
+        $this->title = $this->handleTitle($app);
+        $this->prefix = $this->handlePrefix();
+    }
+
+    private function handleTitle(\WikiaApp $app): string {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 4, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)